### ISSUE TITLE
How do we update pul_base in projects that are subthemed?

### ISSUE THREAD

**Issue by sdellis** [pulibrary/pul_base_deprecated#8]
We need either a simpler process or better documentation (or both) for managing projects with pul_base as a dependency.  To update a sub-themed site to the latest version of pul_base, it seems that one needs to: 
1. Clone pul_base
2. Copy pul_base into the sites/all/themes/ directory of the subthemed drupal project.
3. Reinstall the front-end development environment by running... 
4. bundle install
5. npm install -g grunt-cli
6. npm install

Afterwards, when I make a simple change to one of the sass files while running "grunt watch" a number of files in css/layouts (among others) say they've been changed in addition to the one I changed.  I'm not sure if I'm skipping a step somewhere or have to run some kind of "clean" command.  I can get it to compile with "grunt build --force" but with lots of linting warnings.  I am assuming that this is a problem with the existing sass files from the errors, but it could be a problem with my environment. 

Any code that produces warnings at compilation should not be in the Master branch.  It should only be merged into the master branch after the warnings are resolved and it's stable.

**Comment by kevinreiss** [comment]
Some new ideas on this (thanks to Peter Wolanin). Use git subtree. 

To add the theme's repo to a project

```
git subtree add --squash --prefix=sites/all/themes/pul_base https://github.com/pulibrary/pul_base.git 7.x-1.0-alpha1
```

To update the theme in an existing project (only after it was initially added with the subtree command above).
```
git subtree pull --squash --prefix=sites/all/themes/pul_base https://github.com/pulibrary/pul_base.git 7.x-1.0-my-new-releast-tag
```

This needs further testing to see it will really work effectively. More background reading at:

https://www.acquia.com/blog/using-git-subtree-make-distro-your-docroot
http://blogs.atlassian.com/2013/05/alternatives-to-git-submodule-git-subtree/

I tried in vain to get the https://www.drupal.org/project/erus module to work. Short of reaching out to the module maintainer I'm thinking of abandoning it particular since it seems to have no user base at all. I like the idea of using git better than writing custom Drush commands.

**Comment by sdellis** [comment]
Excellent!  Is there a way to create a custom drush command that has some conventions to make it easier to remember?  Something like:
```drush [theme_name | module_name] add | pull [relase/tag]```
